test/test_base.py

```diff
@@ -11,7 +12,10 @@ class TestBase:
     def __init__(self):
         # Warnet server stdout gets logged here
         self.tmpdir = Path(mkdtemp(prefix="warnet_test_"))
-        self.logfilepath = self.tmpdir / "warnet.log"
+
+        os.environ["XDG_STATE_HOME"] = f"{self.tmpdir}"
```

Comment:
Oh this is pretty lush! I was worried about setting this here (rather than reading it), but setting it is only local to the python process and its children, so this does get propagated to _server.py_ and used, but also doesn't actually set it in your shell. Neat!